src/overlays/actors/ovl_En_Ossan/z_en_ossan.c

```diff
@@ -37,227 +114,2413 @@ const ActorInit En_Ossan_InitVars = {
+      { SI_HEART, -80, 76, -3 } },
+
+    { { SI_ZORA_TUNIC, 50, 52, -20 },
+      { SI_ARROWS_10, 50, 76, -20 },
+      { SI_HEART, 80, 52, -3 },
+      { SI_ARROWS_30, 80, 76, -3 },
+      { SI_DEKU_NUTS_5, -50, 52, -20 },
+      { SI_ARROWS_50, -50, 76, -20 },
+      { SI_FISH, -80, 52, -3 },
+      { SI_RED_POTION_R50, -80, 76, -3 } },
+
+    { { SI_BOMBS_5_R25, 50, 52, -20 },
+      { SI_BOMBS_10, 50, 76, -20 },
+      { SI_BOMBS_20, 80, 52, -3 },
+      { SI_BOMBS_30, 80, 76, -3 },
+      { SI_GORON_TUNIC, -50, 52, -20 },
+      { SI_HEART, -50, 76, -20 },
+      { SI_RED_POTION_R40, -80, 52, -3 },
+      { SI_HEART, -80, 76, -3 } },
+
+    { { SI_19, 50, 52, -20 },
+      { SI_19, 50, 76, -20 },
+      { SI_19, 80, 52, -3 },
+      { SI_19, 80, 76, -3 },
+      { SI_20, -50, 52, -20 },
+      { SI_20, -50, 76, -20 },
+      { SI_20, -80, 52, -3 },
+      { SI_20, -80, 76, -3 } },
+
+    { { SI_GERUDO_MASK, 50, 52, -20 },
+      { SI_ZORA_MASK, 50, 76, -20 },
+      { SI_MASK_OF_TRUTH, 80, 52, -3 },
+      { SI_GORON_MASK, 80, 76, -3 },
+      { SI_SKULL_MASK, -50, 52, -20 },
+      { SI_KEATON_MASK, -50, 76, -20 },
+      { SI_BUNNY_HOOD, -80, 52, -3 },
+      { SI_SPOOKY_MASK, -80, 76, -3 } },
+};
+static EnOssanGetGirlAParams sShopItemReplaceFunc[] = {
+    ShopItemDisp_Default,   ShopItemDisp_Default,    ShopItemDisp_Default, ShopItemDisp_Default,
+    ShopItemDisp_Default,   ShopItemDisp_Default,    ShopItemDisp_Default, ShopItemDisp_Default,
+    ShopItemDisp_Default,   ShopItemDisp_Default,    ShopItemDisp_Default, ShopItemDisp_Default,
+    ShopItemDisp_Default,   ShopItemDisp_Default,    ShopItemDisp_Default, ShopItemDisp_Default,
+    ShopItemDisp_Default,   ShopItemDisp_Default,    ShopItemDisp_Default, ShopItemDisp_Default,
+    ShopItemDisp_Default,   ShopItemDisp_Default,    ShopItemDisp_Default, ShopItemDisp_Default,
+    ShopItemDisp_Default,   ShopItemDisp_Default,    ShopItemDisp_Default, ShopItemDisp_Default,
+    ShopItemDisp_Default,   ShopItemDisp_Default,    ShopItemDisp_Default, ShopItemDisp_SpookyMask,
+    ShopItemDisp_SkullMask, ShopItemDisp_BunnyHood,  ShopItemDisp_Default, ShopItemDisp_ZoraMask,
+    ShopItemDisp_GoronMask, ShopItemDisp_GerudoMask, ShopItemDisp_Default, ShopItemDisp_Default,
+    ShopItemDisp_Default,   ShopItemDisp_Default,    ShopItemDisp_Default, ShopItemDisp_Default,
+    ShopItemDisp_Default,   ShopItemDisp_Default,    ShopItemDisp_Default, ShopItemDisp_Default,
+    ShopItemDisp_Default,   ShopItemDisp_Default,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC83DC.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_U8(targetMode, 2, ICHAIN_CONTINUE),
+    ICHAIN_F32(targetArrowOffset, 500, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ossan/func_80AC854C.s")
+// When selecting an item to buy, this is the position the item moves to
+static Vec3f sSelectedItemPosition[] = { { 17, 58, 30 }, { -17, 58, 30 } };
```

Comment:
```suggestion
static Vec3f sSelectedItemPosition[] = { { 17.0f, 58.0f, 30.0f }, { -17.0f, 58.0f, 30.0f } };
```
Assuming this matches.